src/cpp/transform.hpp

```diff
@@ -0,0 +1,57 @@
+#ifndef TRANSFORM_H
+#define TRANSFORM_H
+
+#include "exceptions.hpp"
+#include "geom.hpp"
+
+struct Affine {
```

Comment:
I'd suggest adding a short description of the structure and how it relates to Grid?